tests/core/test_memory.py

```diff
@@ -101,6 +101,32 @@ def forward(self, x):
         return self.layer2(self.layer1(x))
 
 
+class LazyModel(LightningModule):
+    """ A model which contains lazy layers with unintialized parameters. """
+
+    def __init__(self):
+
+        super().__init__()
+
+        self.layer1 = nn.LazyLinear(5)
+        self.layer2 = nn.LazyLinear(2)
+
+    def forward(self, input):
+
+        return self.layer2(self.layer1(input))
+
+
+def test_lazy_model_summary():
+    """ Test that lazy model summary does not raise an error. """
+
+    lazy_model = LazyModel()
+
+    try:
+        lazy_model.summarize()
+    except RuntimeError:
+        pytest.fail("Unexpected Runtime Error due to Unitialized layers")
```

Comment:
```suggestion
    lazy_model = LazyModel()
    lazy_model.summarize()

```
add an assert for num_parameters = 0

Reply:
Have a look at the following code snippet:
```python

import torch
from torch.nn import UninitializedParameter

print("bias=True")
model = torch.nn.LazyLinear(3, bias=True)

print(isinstance(model.weight, UninitializedParameter))
print(isinstance(model.bias, UninitializedParameter))
print(sum(p.numel() if not isinstance(p, UninitializedParameter) else 0 for p in model.parameters()))


print("bias=False")
model = torch.nn.LazyLinear(3, bias=False)

print(isinstance(model.weight, UninitializedParameter))
print(isinstance(model.bias, UninitializedParameter))
print(sum(p.numel() if not isinstance(p, UninitializedParameter) else 0 for p in model.parameters()))

```
as you can see, the bias, when present, is always initialized (regular parameter). So the numbers you get are correct.

Reply:
Okay I have added the required assert statement. Should I mark this ready for review @awaelchli ? Thanks

Reply:
great!

Reply:
added a note there too to say it's a bug